formats/mdm/MDM_2_4.ddl

```diff
@@ -0,0 +1,515 @@
+     user_ml2b_command_mac_1_2 # user_ml2b_command_mac_3_to_6
+   let user_ml2b_command_udp_port = BEUInt16
+
+   let user_ml2b_signal_data_ip_address_field_1 = BEUInt32
+   let user_ml2b_signal_data_ip_address_field_2 = BEUInt32
+   let user_ml2b_signal_data_ip_address_field_3 = BEUInt32
+   let user_ml2b_signal_data_ip_address_field_4 = BEUInt32
+   let user_ml2b_signal_data_mac_3_to_6 = BEUInt32
+   let user_ml2b_signal_data_mac_1_2 = BEUInt16
+   let user_ml2b_signal_data_mac =
+     user_ml2b_signal_data_mac_1_2 # user_ml2b_signal_data_mac_3_to_6
+   let user_ml2b_signal_data_udp_port = BEUInt16
+
+   let user_ml2b_context_ip_address_field_1 = BEUInt32
+   let user_ml2b_context_ip_address_field_2 = BEUInt32
+   let user_ml2b_context_ip_address_field_3 = BEUInt32
+   let user_ml2b_context_ip_address_field_4 = BEUInt32
+   let user_ml2b_context_mac_3_to_6 = BEUInt32
+   let user_ml2b_context_mac_1_2 = BEUInt16
+   let user_ml2b_context_mac =
+     user_ml2b_context_mac_1_2 # user_ml2b_context_mac_3_to_6
+   let user_ml2b_context_udp_port = BEUInt16
+
+   ^ {| Body_Signal_Port_User_ID = {
+        signal_port_resource_id = signal_port_resource_id,
+        user_id = user_id,
+        user_ml2b_command_ip_address_field_1 = user_ml2b_command_ip_address_field_1,
+        user_ml2b_command_ip_address_field_2 = user_ml2b_command_ip_address_field_2,
+        user_ml2b_command_ip_address_field_3 = user_ml2b_command_ip_address_field_3,
+        user_ml2b_command_ip_address_field_4 = user_ml2b_command_ip_address_field_4,
+        user_ml2b_command_mac = user_ml2b_command_mac,
+        user_ml2b_command_udp_port = user_ml2b_command_udp_port,
+        user_ml2b_signal_data_ip_address_field_1 = user_ml2b_signal_data_ip_address_field_1,
+        user_ml2b_signal_data_ip_address_field_2 = user_ml2b_signal_data_ip_address_field_2,
+        user_ml2b_signal_data_ip_address_field_3 = user_ml2b_signal_data_ip_address_field_3,
+        user_ml2b_signal_data_ip_address_field_4 = user_ml2b_signal_data_ip_address_field_4,
+        user_ml2b_signal_data_mac = user_ml2b_signal_data_mac,
+        user_ml2b_signal_data_udp_port = user_ml2b_signal_data_udp_port,
+        user_ml2b_context_ip_address_field_1 = user_ml2b_context_ip_address_field_1,
+        user_ml2b_context_ip_address_field_2 = user_ml2b_context_ip_address_field_2,
+        user_ml2b_context_ip_address_field_3 = user_ml2b_context_ip_address_field_3,
+        user_ml2b_context_ip_address_field_4 = user_ml2b_context_ip_address_field_4,
+        user_ml2b_context_mac = user_ml2b_context_mac,
+        user_ml2b_context_udp_port = user_ml2b_context_udp_port
+        } |}
+
+-- See 6.3.5: MDM Health Message (Type 5) Specification, Health Status
+-- Field #1 Format
+bitdata Health_Status_Field_1 where
+  alert_type: Alert_Type
+  port_id: uint 7
+  operational_parameter: Operational_Parameter
+
+-- See 6.3.5: MDM Health Message (Type 5) Specification, Health Status
+-- Field #1 Format
```

Comment:
You don't really need to annotate all alternatives in a `bitdata` union with the types, just annotate one of the should be sufficient, and the rest would be inferred.